components_library/boards/lupv_board.py

```diff
@@ -101,6 +102,9 @@ def __init__(
         # The interrupt type is INT_EXT_SUPER
         self.lupio_pic = LupioPIC(pio_addr=0x20002000, int_type = 0x9)
```

Comment:
Could this `0x9` be replace by the proper exception number symbol (like you had in the PIC)?

Reply:
The interrupt codes are defined in `arch/riscv/faults.hh` (https://github.com/darchr/gem5/blob/develop/src/arch/riscv/faults.hh). I'm not sure how to port that over to Python.

Reply:
In that case, you could at least define a python dict to associate the value to a relevant name:

```python
excep_code = {'INT_EXT_SUPER': 9, 'INT_EXT_MACHINE': 10, ...}
```

This way you can use the name, which makes the code more self-explanatory.